DCGAN/readme.md

```diff
@@ -0,0 +1,140 @@
++ This folder will contain all data related to your experiment such as tensorboard logs, images generated during training and training checkpoints.
+
+## 3. Generating Images from Trained Models
+To generated images from trained models, run
+
+```Batchfile
+python generate.py -dataset mnist/cifar -load_path path/to/directory/containing/checkpoint -grid_size n -save_path directory/where/images/are/saved
+```
+
+The arguments used are explained as follows
+
++ `-dataset` requires either `mnist` or `cifar` according to what dataset the model was trained on.
++ `-load_path` requires the path to the training checkpoint to load.
++ `-grid_size` requires integer `n` and will generate n*n images in a grid.
++ `-save_path` requires the path to the directory where the generated images will be saved. If the directory doesn't exist, the script will create it.
+
+To generate images from pre-trained models, download checkpoint files from the Google Drive link given in the `model.txt` file.
+
+## 4. Model Architecture
+### Generator Model
+![generator_image](resources/generator.png)
++ Each Dense and Conv2DTranspose layer is accompanied by LeakyReLU and BatchNorm layer, except the last layer, which has TanH activation.
++ Input is 100-dimensional noise. It is repeatedly upsampled to produce either 28x28x1(MNIST) or 32x32x3(CIFAR-10) image.
+
+### Discriminator Model
+![discriminator_image](resources/discriminator.png)
++ The Conv2D layers are followed by LeakyReLU and Dropout layers.
++ Output is a single number which tells if the image is real or fake/generated.
+
+## 5. Repository Overview
+This repository contains the following files and folders
+
+1. **configs**: This folder contains config.ini files used for specifying the hyperparameters used in experiments.
+
+2. **experiments**: This folder contains data for different runs.
+
+3. **resources**: Contains media for `readme.md`.
+
+4. `data_loader.py`: Contains helper functions that load and preprocess data.
+
+5. `generate.py`: Used to generate and save images from trained models.
+
+6. `model.py`: Contains helper functions that create generator and discriminator models.
+
+7. `model.txt`: Contains google drive links to trained models.
+
+8. `requirements.txt`: Lists dependencies for easy setup in virtual environments.
+
+9. `train.py`: Contains code to train models from scratch.
+
+## 6. Results Obtained
+### a. Generated Images
+Image generated after training model for 30 epochs on MNIST.
+![mnist_generated](resources/generated_image_mnist.png)
+
+Image generated after training model for 100 epochs on CIFAR-10.
+![cifar_generated](resources/generated_image_cifar.png)
+
+### b. Parameters Used
++ Optimizer used is Adam
++ Learning rate 0.0002, beta-1 0.5
++ Trained for 30 epochs(MNIST) and 100 epochs(CIFAR-10)
++ Batch size 32(MNIST) and 128(CIFAR-10)
++ The model uses label flipping (i.e. real images are assigned 0 and fake images are assigned 1)
++ Label smoothing is also used
+
+### c. Loss Curves
```

Comment:
iii. Loss Curves

For consistency with the Contents mentioned above in the Readme.